### PR TITLE
Feature/standard flowable

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/tcp/NettyTCPTransportImpl.java [hunk 1]
```diff
@@ -109,6 +122,7 @@
 			TCPConstants.METADATA_PORT, String.valueOf(port)
 		);
 		this.priority = config.priority(0);
+		this.messageBufferSize = config.messageBufferSize(1);
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/tcp/NettyTCPTransportImpl.java:125]: This default value of 1 seems way too small.  It should be some multiple of the number of validators, so we don't lose vote messages.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/udp/NettyUDPTransportImpl.java [hunk 2]
```diff
@@ -104,6 +114,7 @@
 		this.controlFactory = controlFactory;
 		this.connectionFactory = connectionFactory;
 		this.priority = config.priority(1000);
+		this.messageBufferSize = config.messageBufferSize(255);
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/udp/NettyUDPTransportImpl.java:117]: This is a reasonable default IMO, but it really should be consistent with the TCP transport.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/udp/UDPNettyMessageHandler.java [hunk 3]
```diff
@@ -41,40 +46,68 @@
...
 
+	Flowable<InboundMessage> inboundMessageRx() {
+		return rawMessageSink
+			.onBackpressureBuffer(
+				this.bufferSize,
+				() -> {
+					this.counters.increment(SystemCounters.CounterType.NETWORKING_UDP_DROPPED_MESSAGES);
+					if (droppedMessagesRateLimiter.tryAcquire()) {
+						log.warn("UDP msg buffer overflow, dropping msg");
+					}
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/udp/UDPNettyMessageHandler.java:73]: Would like to see dropped message logs at trace level if rate limiter limits rate.  Something like:
```
Level logLevel = droppedMessagesRateLimiter.tryAcquire() ? Level.WARN : Level.TRACE;
log.log(logLevel, ...);
```

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/tcp/TCPNettyMessageHandler.java [hunk 4]
```diff
@@ -43,42 +48,70 @@
...
-		this.messageSink = messageSink;
+	TCPNettyMessageHandler(SystemCounters counters, int bufferSize) {
+		this.counters = counters;
+		this.bufferSize = bufferSize;
+		this.inboundMessages = rawMessageSink
+			.onBackpressureBuffer(
+				this.bufferSize,
+				() -> {
+					this.counters.increment(CounterType.NETWORKING_TCP_DROPPED_MESSAGES);
+					log.warn("TCP msg buffer overflow, dropping msg");
```
[the remark this answers is not in the document]
Comment (clstrfsck) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/tcp/TCPNettyMessageHandler.java:67]: Also see comments about warn vs trace logging for UDP.

**Review by clstrfsck — CHANGES_REQUESTED** [review]
Some minor comments, but I think the default buffering for TCP messages needs to be larger.

File: radixdlt-core/radixdlt/src/test/java/org/radix/RadixUniverseBuilderTest.java [hunk 5]
```diff
@@ -28,6 +33,40 @@
 import static org.junit.Assert.assertTrue;
 
 public class RadixUniverseBuilderTest {
+	private void startPublisher(String name, PublishProcessor<String> pub, long sleep) {
+		new Thread(() -> {
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/test/java/org/radix/RadixUniverseBuilderTest.java:37]: Also, there is nowhere these threads are stopped, so they will run forever.

File: radixdlt-core/radixdlt/src/test/java/org/radix/RadixUniverseBuilderTest.java [hunk 6]
```diff
@@ -28,6 +33,40 @@
...
+				Thread.sleep(1000);
+				System.out.println("next: " + next);
+			});
+		channels.onNext(publisher1.onBackpressureBuffer(1, () -> {}, BackpressureOverflowStrategy.DROP_LATEST));
+		channels.onNext(publisher2.onBackpressureBuffer(1, () -> {}, BackpressureOverflowStrategy.DROP_LATEST));
+		channels.onNext(publisher3.onBackpressureBuffer(1, () -> {}, BackpressureOverflowStrategy.DROP_LATEST));
+		startPublisher("p1", publisher1, 10);
+		startPublisher("p2", publisher2, 1000);
+		startPublisher("p3", publisher3, 1000);
+		Thread.sleep(1000000);
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/test/java/org/radix/RadixUniverseBuilderTest.java:68]: This test looks unfinished.  There isn't any checks or anything, and seems to run for ~1,000 seconds or circa 20 minutes, which is not great.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/udp/UDPNettyMessageHandler.java [hunk 7]
```diff
@@ -41,40 +46,68 @@
...
 	}
 
+	Flowable<InboundMessage> inboundMessageRx() {
+		return rawMessageSink
+			.onBackpressureBuffer(
+				this.bufferSize,
+				() -> {
+					this.counters.increment(SystemCounters.CounterType.NETWORKING_UDP_DROPPED_MESSAGES);
+					if (droppedMessagesRateLimiter.tryAcquire()) {
+						log.warn("UDP msg buffer overflow, dropping msg");
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/udp/UDPNettyMessageHandler.java:72]: Don't think this information is available

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/tcp/NettyTCPTransportImpl.java (hunk 1 above)
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/tcp/NettyTCPTransportImpl.java:125]: This is a per channel buffer.

**Review by clstrfsck — APPROVED** [review]

**Pull request merged**